x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/mapper/Mapper.java

```diff
@@ -102,7 +103,13 @@ private PhysicalPlan mapUnary(UnaryPlan unary) {
             // 5. So we should be keeping: LimitExec, ExchangeExec, OrderExec, TopNExec (actually OrderExec probably can't happen anyway).
             Holder<Boolean> hasFragment = new Holder<>(false);
 
+            // Remove most plan nodes between this remote ENRICH and the data node's fragment so they're not executed twice;
+            // include the plan up until this ENRICH in the fragment.
             var childTransformed = mappedChild.transformUp(f -> {
+                if (f instanceof BinaryExec be) {
+                    // Remove any LOOKUP JOIN or inline Join from INLINE STATS do avoid double execution
+                    return be.left();
+                }
```

Comment:
This is the main fix: don't duplicate the join node after a remote enrich.

Reply:
Actually, I think there's a saner approach. Any LOOKUP JOIN before a remote ENRICH _is implicitly remote, too_. We should treat them as such, otherwise we will have to redo a lot of validation and look into edge cases that are already solved for remote LOOKUP JOIN.

This approach is implemented in https://github.com/elastic/elasticsearch/pull/131286/commits/0f65dd57636c86a753bfa7f11e5e8a15597bc883.